src/runtime/Types/ClassDerived.cs

```diff
@@ -867,18 +865,11 @@ public class PythonDerivedType
                 throw new NullReferenceException("Instance must be specified when getting a property");
             }
 
-            PyGILState gs = Runtime.PyGILState_Ensure();
-            try
-            {
-                using var pyself = new PyObject(self.CheckRun());
-                using (PyObject pyvalue = pyself.GetAttr(propertyName))
-                {
-                    return pyvalue.As<T>();
-                }
-            }
-            finally
+            using var _ = new Py.GILState();
+            using var pyself = new PyObject(self.CheckRun());
+            using (PyObject pyvalue = pyself.GetAttr(propertyName))
             {
-                Runtime.PyGILState_Release(gs);
+                return pyvalue.As<T>();
```

Comment:
Not coming from your change, but disposing `PyObject`, that was used in `As<T>` operation is semantically incorrect since `As<T>` can store that `PyObject` instance elsewhere.

Reply:
NVM, that was wrong. Currently `As<T>` takes this as a `BorrowedReference`.